### ISSUE TITLE
Automate Tests

### ISSUE THREAD

**Issue by Edenhofer** [davidogara/hetGPy#11]
It would be great if the test and ideally also the examples would be run automatically in e.g. a GitHub CI.

Part of https://github.com/openjournals/joss-reviews/issues/7518 .

**Comment by davidogara** [comment]
Thanks for bringing this up @Edenhofer -- I pushed a series of commits that do the following:

Run the tests and examples in a Github CI:
- Added a new workflow (`.github/workflows/run-main-tests-on-push.yaml`) that runs the smaller integration tests (ones that run in 1-2 seconds and compare against known benchmarks). This runs on any push to the main branch.
- The CI also runs the first two example notebooks (since they only take a few seconds to run).

I also added another workflow run-tests.yaml to:
- Run the entire suite of tests. For example, some tests require running R routines to compare to the `hetGP` R package, which take a bit longer, so they are run less frequently.
- They also run all of the example notebooks (some of which take a few minutes to run).
- This one requires manual activation (but could be changed to run on releases or something similar)
